### ISSUE TITLE
Can csi snapshoptter be used in velero backup?

### ISSUE THREAD

**Issue by liyimeng** [openebs/zfs-localpv#397]
**Describe the problem/challenge you have**
[A description of the current limitation/problem/challenge that you are experiencing.]

Try to keep it simple

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
velero can use CSI Snapshot for backup according to this https://velero.io/blog/csi-integration/
can zfs-localpv use this feature without needing extral plugin? 
To my understanding, nfs-localpv is already support CSI Snapshot, it should be a simple work, right?
 
**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**
- ZFS-LocalPV version
- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Comment by pawanpraka1** [comment]
@liyimeng yes, CSI Snapshot can be used for backup.  We can use the velero CSI plugin for backing up the data. Just a note here, the data will be backed up locally only as it will be backed up as a zfs snapshot.

**Comment by ianb-mp** [comment]
I'm interested in this question. I'd like to be able to take backups and volume snapshots of Kubevirt VMs and have them uploaded to S3

According to this https://github.com/vmware-tanzu/velero/issues/3544 it should be supported in Velero, but unsure if zfs-localpv supports it on the backend.

I tried using the CSI snapshot with https://github.com/vmware-tanzu/velero-plugin-for-csi and while the volume snapshot did succeed, I couldn't get it to upload to S3 (velero backup would get stuck in `WaitingForPluginOperations` phase).

I've ended up using https://github.com/openebs/velero-plugin instead of `velero-plugin-for-csi` and that does what I need, however I'm concerned that velero-plugin may be deprecated - see: https://github.com/openebs/zfs-localpv/issues/511

**Comment by Abhinandan-Purkait** [comment]
@ianb-mp  `https://github.com/openebs/velero-plugin` will not be deprecated. It will be considered for development and enhancements.

**Comment by Abhinandan-Purkait** [comment]
@w3aman can you provide your insights on this?

**Comment by sinhaashish** [comment]
> I'm interested in this question. I'd like to be able to take backups and volume snapshots of Kubevirt VMs and have them uploaded to S3
> 
> According to this [vmware-tanzu/velero#3544](https://github.com/vmware-tanzu/velero/issues/3544) it should be supported in Velero, but unsure if zfs-localpv supports it on the backend.
> 
> I tried using the CSI snapshot with https://github.com/vmware-tanzu/velero-plugin-for-csi and while the volume snapshot did succeed, I couldn't get it to upload to S3 (velero backup would get stuck in `WaitingForPluginOperations` phase).
> 
> I've ended up using https://github.com/openebs/velero-plugin instead of `velero-plugin-for-csi` and that does what I need, however I'm concerned that velero-plugin may be deprecated - see: [#511](https://github.com/openebs/zfs-localpv/issues/511)

@ianb-mp   CSI Snapshot can be used for backup. To upload it to s3 you need the plugin for specific cloud provider. like https://github.com/vmware-tanzu/velero-plugin-for-aws for aws  and https://github.com/vmware-tanzu/velero-plugin-for-gcp for gcp .  I would suggest to use cloud specific plugin.
